### PR TITLE
feat(book): `op-reth` runbook

### LINKED ISSUE #5432: feat: `op-reth` "how to run" docs in the book
## Overview

Add a section to the `reth` book that describes how to set up an `op-reth` node for an OP Stack chain without legacy state.

**Rough Reference Gist**
https://gist.github.com/clabby/565b4645cfa65b66396c6141e3567ad3

**Periphery Information**
* It should be clearly outlined that chains with pre-Bedrock state, i.e. OP Mainnet and OP Goerli, cannot be synced currently due to the lack of a database migration tool.
* The `op-node` has a feature, `rethdb`, that can allow it to read receipts directly from a L1 reth DB that is on the same machine. The guide should point out that while this is not required, it can speed up sync times if applicable to the user's environment.
* Base node information can be found in [Base's `node` repository](github.com/base-org/node)

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by clabby**
## Overview

Adds a page to the `reth` book describing high-level differences between `op-reth` and `reth`, links to further specifications, and a "how to" section for running `op-reth` nodes.

**Metadata**
closes #5432

### CODE REVIEW

File: book/run/optimism.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
+# Running Reth on OP Stack chains
+
+`reth` ships with the `optimism` feature flag, enabling support for OP Stack chains out of the box. Optimism has a small diff from the [L1 EELS][l1-el-spec], comprising of the following key changes:
```
Comment (refcell) [inline book/run/optimism.md:3]: I think it's worthwhile to specify reth **crates** and not the **reth** binary here

File: book/run/optimism.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+1. A new transaction type, [`0x7E (Deposit)`][deposit-spec], which is used to deposit funds from L1 to L2.
+1. Modifications to the `PayloadAttributes` that allow the [sequencer][sequencer] to submit transactions in-order to the EL through the Engine API.
```
Comment (refcell) [inline book/run/optimism.md:5]: `in-order` reads slightly ambiguous to me, can we say `first-in first-out (FIFO)` instead or something to that effect?

File: book/run/optimism.md [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+This will build the `rethdb-reader` dylib and instruct the `op-node` build to statically link this dylib into the binary. The `op-node` binary will be installed to `/usr/bin/op-node`.
+
+### Running `op-reth`
+
+The `optimism` feature flag in `op-reth` adds several new CLI flags to the `reth` binary:
+1. `--rollup.sequencer-http <uri>` - The sequencer endpoint to connect to. Transactions sent to the `op-reth` EL are also forwarded to this sequencer endpoint for inclusion, as the sequencer is the entity that builds blocks on OP Stack chains.
+1. `--rollup.disable-tx-pool-gossip` - Disables gossiping of transactions in the mempool to peers. This can be ommitted for personal nodes, though providers should always opt to enable this flag.
+1. `--rollup.enable-genesis-walkback` - Disables setting the forkchoice status to tip on startup, making the `op-node` walk back to genesis and verify the integrity of the chain before starting to sync. This can be ommitted unless a corruption of local chainstate is suspected.
+
+For this example, we'll start a `Base Mainnet` node.
```
Comment (refcell) [inline book/run/optimism.md:67]: Let's put this at the top of the `## Running on Optimism` section so it's clear to people what they'll be running from the start.

**Review by refcell — CHANGES_REQUESTED** [review]

File: book/run/optimism.md (hunk 2 above)
Comment (clabby) [reply book/run/optimism.md]: Agree - exhaustively describing the behavior of sequencer tx ordering is a bit verbose for here, though, imo. What do you think about the update?

Comment (refcell) [reply book/run/optimism.md]: Discussed offline, once updated gtg 👍

**Review by refcell — APPROVED** [review]
Reads well to me now, nice work 💯

**Review by gakonst — APPROVED** [review]
good w me beyond my dm'ed comments

**Review by mattsse — APPROVED** [review]
lgtm, pending @shekhirin

File: book/run/optimism.md [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+    --rpc.port=7000 \
+    --l1.trustrpc
+```
+
+If you opted to build the `op-node` with the `rethdb` build tag, this "`RPCKind`" can be enabled via appending two extra flags to the `op-node` invocation:
+```sh
+op-node \
+    # ...
+    --l1.rpckind=reth_db \
+    --l1.rethdb=<your_L1_reth_db_path>
```
Comment (shekhirin) [inline book/run/optimism.md:102]: is this a reth datadir, `db` dir inside it or `mdbx.dat` file path?

**Review by shekhirin — APPROVED** [review]
LGTM modulo one clarification

File: book/run/optimism.md (hunk 4 above)
Comment (clabby) [reply book/run/optimism.md:105]: Added a clarification

Comment (shekhirin) [reply book/run/optimism.md:105]: I think doing `--l1.rethdb=$(reth db path)` would be confusing, but it's the easiest way to get the path.

Comment (clabby) [reply book/run/optimism.md:106]: Done 👍

Comment (shekhirin) [reply book/run/optimism.md:106]: sick

**Pull request merged**